vello_encoding/src/ramp_cache.rs

```diff
@@ -104,7 +125,17 @@ fn make_ramp(stops: &[ColorStop]) -> impl Iterator<Item = u32> + '_ {
         let c = if du < 1e-9 {
             this_c
         } else {
-            last_c.lerp(this_c, (u - last_u) / du, HueDirection::default())
+            match interpolation_alpha_space {
+                InterpolationAlphaSpace::Premultiplied => {
+                    last_c.lerp(this_c, (u - last_u) / du, HueDirection::default())
+                }
+                InterpolationAlphaSpace::Unpremultiplied => {
+                    AlphaColor::<Srgb>::new(std::array::from_fn(|i| {
+                        last_c.components[i]
+                            + (this_c.components[i] - last_c.components[i]) * ((u - last_u) / du)
+                    }))
```

Comment:
NOTE: lerp on AlphaColor in color converts colors to premultiplied does interpolation and then unpremultiplied (interpolation happens in premultiplied alpha space).